src/functions-and-arrays.js

```diff
@@ -1,41 +1,86 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  if (a > b) {
+    return a
+  }
+  else if (a < b) {
+    return b
+  }
+  else if (a === b) {
+    return a, b
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(a) {
+  if (a.length === 0) {
+    return null
+  }
+  else if (a.length === 1) {
+    return a[0]
+  }
+
+  //  Imagino que se podria hacer usando un for para comparar las listas, pero no me da la cabeza
+  return
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(a) {
+  let sum = a.reduce(function (a, b) {
+    return a + b;
+  }, 0)
+  return sum
+
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(a) {
+  let sum = a.reduce(function (a, b) {
+    return a + b;
+  }, 0)
+  return sum
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(a) {
+  let total = 0
+  if (a.length === 0) {
+    return null
+  }
+  for (var i = 0; i < a.length; i++) {
+    total += a[i];
+  }
+  let avg = total / a.length;
+  return avg
+}
+
```

Comment:
Aqui podías reutilizar la función de sumar para no repetir codigo.

```suggestion
function averageNumbers(a) {
  let total = sumNumbers(a)
  let avg = total / a.length;
  return avg
}

```